business/checkers/serviceentries/workload_entry_address_match.go

```diff
@@ -0,0 +1,77 @@
+package serviceentries
+
+import (
+	"fmt"
+
+	"istio.io/client-go/pkg/apis/networking/v1alpha3"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/kiali/kiali/models"
+)
+
+type HasMatchingWorkloadEntryAddress struct {
+	ServiceEntry    v1alpha3.ServiceEntry
+	WorkloadEntries map[string][]string
+}
+
+const MeshInternal = 1
+
+func (in HasMatchingWorkloadEntryAddress) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	if in.ServiceEntry.Spec.Location != MeshInternal {
+		return validations, true
+	}
+
+	if in.ServiceEntry.Spec.WorkloadSelector == nil {
+		return validations, true
+	}
+
+	var targetAddresses []string
+	seSelector := labels.Set(in.ServiceEntry.Spec.WorkloadSelector.Labels).AsSelector()
+
+	for labelsMap, weAddressMap := range in.WorkloadEntries {
+		workloadLabelsSet, err := labels.ConvertSelectorToLabelsMap(labelsMap)
+		if err != nil {
+			continue
+		}
+
+		if seSelector.Matches(workloadLabelsSet) {
+			targetAddresses = append(targetAddresses, weAddressMap...)
+		}
+	}
+
+	if targetAddresses == nil {
```

Comment:
Shouldn't there be a validation warning message, something like "No matching workload selector was found" ? Because this can be a gap in validation.

Reply:
I think it would be part of another validation. As far as I understood, ServiceEntries doesn't necessarily need workloads to work. You need workloads when you have a MESH_INTERNAL and not using the `endpoints` field (but the address).

So the validation you suggest should be a bit more specific and we should double check if I am right with the previous assumptions.

Thanks!